libexec/python/tests/test_core.py

```diff
@@ -149,18 +149,18 @@ def test_parse_image_uri(self):
         self.assertTrue(digest['namespace'] == 'mix')
 
         print("Case 8: Custom uri should use it.")
-        image_name = "catdog://meow/mix/%s:%s" % (self.repo_name,
-                                                  self.tag)
+        image_name = "catdog://meow/mix/tenders/%s:%s" % (self.repo_name,
+                                                          self.tag)
         digest = parse_image_uri(image_name, uri="catdog://")
         self.assertTrue(digest['registry'] == 'meow')
-        self.assertTrue(digest['namespace'] == 'mix')
+        self.assertTrue(digest['namespace'] == 'mix/tenders')
 
         print("Case 9: Digest version should be parsed")
-        image_name = ("catdog://meow/mix/%s:%s@sha:256xxxxxxxxxxxxxxx"
+        image_name = ("catdog://meow/mix/original/choice/%s:%s@sha:256xxxxxxxxxxxxxxx"  # noqa
                       % (self.repo_name, self.tag))
         digest = parse_image_uri(image_name, uri="catdog://")
         self.assertTrue(digest['registry'] == 'meow')
-        self.assertTrue(digest['namespace'] == 'mix')
+        self.assertTrue(digest['namespace'] == 'mix/original/choice')
```

Comment:
Just to be clear - you are creating images with names like `mix/original/choice` ? So you would do `docker run -it mix/original/choice bash` ? I haven't seen this kind of usage before, and it's a bit challenging because the entire parsing of the namespace is dependent on the slashes.